src/charm.py

```diff
@@ -41,6 +43,36 @@
 CREATED_BY_LABEL = "gateway-api-integrator.charm.juju.is/managed-by"
 
 
+@lru_cache(maxsize=1)
```

Comment:
Makes sense, I was putting it here in case of deferred events and just to be sure that the client is not initialized multiple times, I'll remove the cache decorator.